iothub/device/src/Authentication/AuthenticationWithTokenRefresh.cs

```diff
@@ -32,20 +32,20 @@ public abstract class AuthenticationWithTokenRefresh : IAuthenticationMethod
         /// The time buffer before expiry when the token should be renewed, expressed as a percentage of the time to live.
         /// The default behavior is that the token will be renewed when it has <see cref="DefaultSasRenewalBufferPercentage"/> percent or less of its lifespan left.
         ///</param>
-        /// <exception cref="ArgumentOutOfRangeException">Thrown if <paramref name="suggestedTimeToLive"/> is a negative timespan, or if
```

Comment:
I could see an argument for keeping this type of exception. It is more specific.

Reply:
It inherits from ArgumentException, btw.

Reply:
Yeah, throwing more specifically is better than maintaining a shorter list of exceptions we have in such a case. Sure, I will change it back.